### PR TITLE
Add MPNet Model

### PR DESCRIPTION
**Pull request by JunnYu**
飞桨论文复现挑战赛（第四期）MPNet: Masked and Permuted Pre-training for Language Understanding 论文复现提交。

### CODE REVIEW

File: examples/language_model/mpnet/README.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
...ssful pre-training models. Since BERT neglects dependency among predicted tokens, XLNet introduces permuted language modeling (PLM) for pretraining to address this problem. However, XLNet does not leverage the full position information of a sentence and thus suffers from position discrepancy between pre-training and fine-tuning. In this paper, we propose MPNet, a novel pre-training method that inherits the advantages of BERT and XLNet and avoids their limitations. MPNet leverages the dependency among predicted tokens through permuted language modeling (vs. MLM in BERT), and takes auxiliary position information as input to make the model see a full sentence and thus reducing the position discrepancy (vs. PLM in XLNet). We pre-train MPNet on a large-scale dataset (over 160GB text corpora) and fine-tune on a variety of down-streaming tasks (GLUE, SQuAD, etc). Experimental results show that MPNet outperforms MLM and PLM by a large margin, and achieves better results on these tasks compared with previous state-of-the-art pre-trained methods (e.g., BERT, XLNet, RoBERTa) under the same model setting. The code and the pre-trained models are available at: https://github.com/microsoft/MPNet.
```
Comment (gongel) [inline examples/language_model/mpnet/README.md]: Please translate to Chinese.

File: examples/language_model/mpnet/README.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+<p align="center">
+    <img src="./figure/QQP.png" width="100%" />
+</p>
+<p align="center">
+    <img src="./figure/SQuAD.png" width="100%" />
+</p>
+
+## 快速开始
+
+### 模型精度对齐
```
Comment (gongel) [inline examples/language_model/mpnet/README.md]: Please remove.

File: examples/language_model/mpnet/compare.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (gongel) [inline examples/language_model/mpnet/compare.py]: Please delete `compare.py`

**Review by gongel — CHANGES_REQUESTED** [review]
Please add docstrings for all your classes and methods which might be utilized by users. You can refer to paddlenlp.transformers.bert.

File: paddlenlp/transformers/mpnet/tokenizer.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    Construct a MPNet tokenizer. `MPNetTokenizer` is almost identical to `BertTokenizer` and runs end-to-end 
+    tokenization: punctuation splitting and wordpiece. Refer to superclass `BertTokenizer` for usage examples 
+    and documentation concerning parameters.
+    """
+    resource_files_names = {"vocab_file": "vocab.txt"}  # for save_pretrained
+    pretrained_resource_files_map = {
+        "vocab_file": {
+            "mpnet-base":
+            "https://paddlenlp.bj.bcebos.com/models/transformers/mpnet/mpnet-base/vocab.txt",
```
Comment (gongel) [inline paddlenlp/transformers/mpnet/tokenizer.py:30]: 词表貌似没有ready？

Comment (gongel) [reply paddlenlp/transformers/mpnet/tokenizer.py:30]: 另外Tokenizer默认会返回`token_type_ids`，但是这个在`MPNet`是不需要的

File: paddlenlp/transformers/mpnet/modeling.py [hunk 5]
```diff
@@ -0,0 +1,904 @@
...
+                              ).astype(input_ids.dtype)
+
+        if attention_mask.ndim == 2:
+            extended_attention_mask = attention_mask.unsqueeze(axis=[1, 2])
+            extended_attention_mask = (1.0 - extended_attention_mask) * -10000.0
+
+        embedding_output = self.embeddings(input_ids, position_ids)
+
+        encoder_outputs, _ = self.encoder(embedding_output,
+                                          extended_attention_mask)
```
Comment (gongel) [inline paddlenlp/transformers/mpnet/modeling.py]: `extended_attention_mask `--->`attention_mask `

File: paddlenlp/transformers/mpnet/modeling.py [hunk 6]
```diff
@@ -0,0 +1,904 @@
...
+class MPNetForTokenClassification(MPNetPretrainedModel):
+    """
+    MPNet Model with a token classification head on top (a linear layer on top of the hidden-states output) e.g.
+    for Named-Entity-Recognition (NER) tasks.
+
+    Args:
+        mpnet (:class:`MPNetModel`):
+            An instance of MPNetModel.
+        num_classes (int, optional):
+            The number of classes. Defaults to `2`.
```
Comment (gongel) [inline paddlenlp/transformers/mpnet/modeling.py:778]: 下面的默认值没给哈

**Review by gongel — CHANGES_REQUESTED** [review]
Thanks for your contribution and please give us feedback.

File: paddlenlp/transformers/mpnet/tokenizer.py (hunk 4 above)
Comment (yingyibiao) [reply paddlenlp/transformers/mpnet/tokenizer.py:30]: > 词表貌似没有ready？

done

**Review by yingyibiao — COMMENTED** [review]
图片可以删了哈，保留表格就好

**Comment by JunnYu** [comment]
好的，过会改

File: paddlenlp/transformers/mpnet/modeling.py (hunk 5 above)
Comment (JunnYu) [reply paddlenlp/transformers/mpnet/modeling.py]: done

File: paddlenlp/transformers/mpnet/modeling.py (hunk 6 above)
Comment (JunnYu) [reply paddlenlp/transformers/mpnet/modeling.py:778]: done

File: paddlenlp/transformers/mpnet/tokenizer.py (hunk 4 above)
Comment (JunnYu) [reply paddlenlp/transformers/mpnet/tokenizer.py:30]: done

File: paddlenlp/transformers/mpnet/tokenizer.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        """
+
+        sep = [self.sep_token_id]
+        cls = [self.cls_token_id]
+
+        if token_ids_1 is None:
+            return len(cls + token_ids_0 + sep) * [0]
+        return len(cls + token_ids_0 + sep + sep + token_ids_1 + sep) * [0]
+
+    def __call__(self,
```
Comment (gongel) [inline paddlenlp/transformers/mpnet/tokenizer.py]: 这个放在`def __init__`后面吧

**Review by gongel — DISMISSED** [review]
处理下冲突

File: paddlenlp/transformers/mpnet/tokenizer.py (hunk 7 above)
Comment (JunnYu) [reply paddlenlp/transformers/mpnet/tokenizer.py]: done

**Review by gongel — DISMISSED** [review]
LGTM😊

File: paddlenlp/transformers/mpnet/modeling.py [hunk 8]
```diff
@@ -0,0 +1,903 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+# Copyright 2018 The Google AI Language Team Authors and The HuggingFace Inc. team.
```
Comment (gongel) [inline paddlenlp/transformers/mpnet/modeling.py]: Google AI Language Team？

Comment (JunnYu) [reply paddlenlp/transformers/mpnet/modeling.py]: 我偷懒从别的文件复制的，待会改

**Review by gongel — APPROVED** [review]
